torch/fx/symbolic_trace.py

```diff
@@ -330,7 +346,18 @@ def proxy_placeholder(name: str):
                 args.append(proxy_placeholder('**' + next(names_iter)))
             root_fn = _patch_function(root_fn, len(args))
 
-        return root_fn, args
+        flat_args, in_spec = pytree.tree_flatten(tuple(args))
+        if in_spec.num_leaves == len(args) and all(isinstance(i, pytree.LeafSpec) for i in in_spec.children_specs):
+            return root_fn, args
+
+        self.graph._in_spec = in_spec
+        self.graph._orig_args = orig_args[:total_args]
+
+        def flatten_fn(*args):
+            tree_args = pytree.tree_unflatten(list(args), in_spec)
+            return root_fn(*tree_args)
+
+        return flatten_fn, flat_args
```

Comment:
style nit: put the common (non-pytree) case in the unconditional path and put the pytree case in a branch. Also comment on what's happening, e.g. 

```
        flat_args, in_spec = pytree.tree_flatten(tuple(args))

        if in_spec.num_leaves != len(args) or not all(isinstance(i, pytree.LeafSpec) for i in in_spec.children_specs):
            # In the case that we have pytree-flattened inputs in `concrete_args`, generate a flattening wrapper
            # around the original root function and return that
            self.graph._in_spec = in_spec
            self.graph._orig_args = orig_args[:total_args]

            def flatten_fn(*args):
                tree_args = pytree.tree_unflatten(list(args), in_spec)
                return root_fn(*tree_args)

            return flatten_fn, flat_args

        return root_fn, args
```